contrib/raftexample/kvstore.go

```diff
@@ -33,12 +34,15 @@ type kv struct {
 	Val string
 }
 
-func newKVStore(proposeC chan<- string, commitC <-chan *string, errorC <-chan error) *kvstore {
+func newKVStore(proposeC chan<- string, commitC <-chan *string, errorC <-chan error, ready chan struct{}) *kvstore {
```

Comment:
One extra remark on this approach-- it won't know the snapIndex if it's using nil commits. It can still work without passing that information around but it'll need to be careful about serialization and write ordering (e.g., something like writing a snapshot to a new file, then renaming over the old file) .